File: Prac 3 Python/dataset_handler.py
import pandas as pd
import requests
from io import StringIO

class DatasetHandler:
    def __init__(self):
        self.data = None
    
    def create_sample_dataset(self, filename='phishing_dataset.csv'):
        """Create a sample dataset for demonstration"""
        # Sample data with more realistic examples
        data = {
            'url': [
                # Legitimate URLs
                'https://www.google.com',
                'https://www.github.com',
                'https://www.stackoverflow.com',
                'https://www.wikipedia.org',
                'https://www.microsoft.com',
                'https://www.apple.com',
                'https://www.amazon.com',
                'https://www.facebook.com',
                'https://www.twitter.com',
                'https://www.linkedin.com',
                'https://www.youtube.com',
                'https://www.netflix.com',
                'https://www.reddit.com',
                'https://www.instagram.com',
                'https://www.pinterest.com',
                
                # Phishing URLs (simulated examples)
                'http://secure-paypal-update.com/login',
                'https://amazon-security-alert.net/confirm',
                'http://192.168.1.1/bank-login',
                'https://paypal-verification.suspicious-domain.com',
                'http://bank-account-update.fake-site.org',
                'https://secure-login-ebay.malicious.com',
                'http://confirm-account-amazon.phishing.net',
                'https://update-paypal-security.fake.org',
                'http://bank-alert-signin.suspicious.com',
                'https://account-verification-paypal.malware.net',
                'http://secure-bank-login.phishing-site.com',
                'https://paypal-account-limited.fake.org',
                'http://amazon-account-suspended.malicious.net',
                'https://ebay-security-notice.suspicious.com',
                'http://bank-verification-required.phishing.org'
            ],
            'label': [0] * 15 + [1] * 15  # 0 = Legitimate, 1 = Phishing
        }
        
        df = pd.DataFrame(data)
        df.to_csv(filename, index=False)
        print(f"Sample dataset created: {filename}")
        return df
    
    def load_dataset(self, filename):
        """Load dataset from CSV file"""
        try:
            self.data = pd.read_csv(filename)
            print(f"Dataset loaded: {filename}")
            print(f"Shape: {self.data.shape}")
            return self.data
        except FileNotFoundError:
            print(f"File {filename} not found. Creating sample dataset...")
            return self.create_sample_dataset(filename)
    
    def get_urls_and_labels(self):
        """Extract URLs and labels from the dataset"""
        if self.data is None:
            raise ValueError("No dataset loaded. Please load a dataset first.")
        
        urls = self.data['url'].tolist()
        labels = self.data['label'].tolist()
        return urls, labels
    
    def dataset_info(self):
        """Display dataset information"""
        if self.data is None:
            print("No dataset loaded.")
            return
        
        print("Dataset Information:")
        print(f"Total URLs: {len(self.data)}")
        print(f"Legitimate URLs: {sum(self.data['label'] == 0)}")
        print(f"Phishing URLs: {sum(self.data['label'] == 1)}")
        print("\nSample URLs:")
        print(self.data.head())

if __name__ == "__main__":
    handler = DatasetHandler()
    dataset = handler.load_dataset('phishing_dataset.csv')
    handler.dataset_info()

File: Prac 3 Python/DatasetHandler.java
package com.phishing.util;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class DatasetHandler {
    private List<String> urls;
    private List<Integer> labels;
    
    public DatasetHandler() {
        createSampleDataset();
    }
    
    /**
     * Creates a sample dataset with legitimate and phishing URLs
     */
    private void createSampleDataset() {
        urls = new ArrayList<>();
        labels = new ArrayList<>();
        
        // Legitimate URLs
        List<String> legitimateUrls = Arrays.asList(
            "https://www.google.com",
            "https://www.github.com",
            "https://www.stackoverflow.com",
            "https://www.wikipedia.org",
            "https://www.microsoft.com",
            "https://www.apple.com",
            "https://www.amazon.com",
            "https://www.facebook.com",
            "https://www.twitter.com",
            "https://www.linkedin.com",
            "https://www.youtube.com",
            "https://www.netflix.com",
            "https://www.reddit.com",
            "https://www.instagram.com",
            "https://www.pinterest.com"
        );
        
        // Phishing URLs (simulated examples)
        List<String> phishingUrls = Arrays.asList(
            "http://secure-paypal-update.com/login",
            "https://amazon-security-alert.net/confirm",
            "http://192.168.1.1/bank-login",
            "https://paypal-verification.suspicious-domain.com",
            "http://bank-account-update.fake-site.org",
            "https://secure-login-ebay.malicious.com",
            "http://confirm-account-amazon.phishing.net",
            "https://update-paypal-security.fake.org",
            "http://bank-alert-signin.suspicious.com",
            "https://account-verification-paypal.malware.net",
            "http://secure-bank-login.phishing-site.com",
            "https://paypal-account-limited.fake.org",
            "http://amazon-account-suspended.malicious.net",
            "https://ebay-security-notice.suspicious.com",
            "http://bank-verification-required.phishing.org"
        );
        
        // Add legitimate URLs with label 0
        for (String url : legitimateUrls) {
            urls.add(url);
            labels.add(0); // 0 = Legitimate
        }
        
        // Add phishing URLs with label 1
        for (String url : phishingUrls) {
            urls.add(url);
            labels.add(1); // 1 = Phishing
        }
    }
    
    /**
     * Gets the list of URLs in the dataset
     * @return List of URLs
     */
    public List<String> getUrls() {
        return urls;
    }
    
    /**
     * Gets the list of labels in the dataset
     * @return List of labels (0 = Legitimate, 1 = Phishing)
     */
    public List<Integer> getLabels() {
        return labels;
    }
    
    /**
     * Gets dataset information as a formatted string
     * @return Dataset information
     */
    public String getDatasetInfo() {
        int total = urls.size();
        int legitimate = 0;
        int phishing = 0;
        
        for (Integer label : labels) {
            if (label == 0) {
                legitimate++;
            } else {
                phishing++;
            }
        }
        
        StringBuilder info = new StringBuilder();
        info.append("Dataset Information:\n");
        info.append("Total URLs: ").append(total).append("\n");
        info.append("Legitimate URLs: ").append(legitimate).append("\n");
        info.append("Phishing URLs: ").append(phishing).append("\n");
        
        return info.toString();
    }
}